src/vuln_analysis/tools/transitive_code_search.py

```diff
@@ -53,29 +55,34 @@ class PackageAndFunctionLocatorToolConfig(FunctionBaseConfig, name="package_and_
     Package and function locator tool used to validate package names and find function names using fuzzy matching.
     """
 
-
-def get_call_of_chains_retriever(documents_embedder, si):
+def get_call_of_chains_retriever(documents_embedder, si, query: str):
     documents: list[Document]
     git_repo = None
+    code_source_info: SourceDocumentsInfo
     for source_info in si:
         if source_info.type == "code":
+            code_source_info = source_info
             git_repo = documents_embedder.get_repo_path(source_info)
             documents = documents_embedder.collect_documents(source_info)
     if git_repo is None:
         raise ValueError("No code source info found")
-    with open(os.path.join(git_repo, 'ecosystem_data.txt'), 'r', encoding='utf-8') as file:
-        ecosystem = file.read()
-    ecosystem = Ecosystem[ecosystem]
-    coc_retriever = ChainOfCallsRetriever(documents=documents, ecosystem=ecosystem, manifest_path=git_repo)
+    ecosystem = TransitiveCodeSearcher.get_ecosystem(git_repo)
+    coc_retriever = get_chain_of_calls_retriever(ecosystem,
+                                                 documents,
+                                                 git_repo,
+                                                 query,
+                                                 code_source_info)
     return coc_retriever
 
-
-def get_transitive_code_searcher():
+def get_transitive_code_searcher(query: str):
     state: AgentMorpheusEngineState = ctx_state.get()
-    if state.transitive_code_searcher is None:
+    if state.transitive_code_searcher is None or isinstance(state.transitive_code_searcher.chain_of_calls_retriever, JavaChainOfCallsRetriever):
         si = state.original_input.input.image.source_info
         documents_embedder = DocumentEmbedding(embedding=None)
-        coc_retriever = get_call_of_chains_retriever(documents_embedder, si)
+        if state.transitive_code_searcher is not None and isinstance(state.transitive_code_searcher.chain_of_calls_retriever, JavaChainOfCallsRetriever):
+            coc_retriever = get_call_of_chains_retriever(documents_embedder, si, query)
+        else:
+            coc_retriever = get_call_of_chains_retriever(documents_embedder, si, query)
```

Comment:
@tmihalac The if And else parts are identical, so why do you need that if statement at all?